flaml/autogen/agentchat/responsive_agent.py

```diff
@@ -373,27 +510,16 @@ def generate_reply(
             str or dict or None: reply. None if no reply is generated.
         """
         assert messages is not None or sender is not None, "Either messages or sender must be provided."
-        if messages is None:
-            messages = self._oai_messages[sender.name]
-        message = messages[-1]
-        if "function_call" in message:
-            _, func_return = self.execute_function(message["function_call"])
-            return func_return
-        if self._code_execution_config is False:
-            return default_reply if self.llm_config is False else self._oai_reply(messages)
-        code_blocks = extract_code(message["content"])
-        if len(code_blocks) == 1 and code_blocks[0][0] == UNKNOWN:
-            # no code block is found, lang should be `UNKNOWN`
-            if self.llm_config is False:
-                return default_reply
-            # code_blocks, _ = find_code(messages, sys_msg=self._oai_system_message, **self.llm_config)
-            # if len(code_blocks) == 1 and code_blocks[0][0] == UNKNOWN:
-            #     return code_blocks[0][1]
-            return self._oai_reply(messages)
-        # try to execute the code
-        exitcode, logs = self.execute_code_blocks(code_blocks)
-        exitcode2str = "execution succeeded" if exitcode == 0 else "execution failed"
-        return f"exitcode: {exitcode} ({exitcode2str})\nCode output: {logs}"
+        final, reply = self._check_termination_and_human_reply(sender=sender)
```

Comment:
Could we run the registered functions before the default `_check_termination_and_human_reply`? In my case, I need to check the received message, but when human_input_mode=="ALWAYS", the `_check_termination_and_human_reply` will return final as True, and my registered function will not be executed. I still need to override `generate_reply`. But if we run the registered functions first, I can achieve my goal without override the `generate_reply`.

Reply:
If you skip human input by pressing Enter, `_check_termination_and_human_reply` should return final as False unless the msg is a termination msg.